### PR TITLE
Update JuMP and MathOptInterface deps

### PR DESCRIPTION
**Pull request by davidanthoff**
~This first requires https://github.com/chkwon/PATHSolver.jl/issues/60 to be resolved, but then it all seems to work.~

Should be ready now. I also removed a compat entry that declared this as working with Julia 2.0. I believe packages shouldn't do that at this stage, as they aren't known to work with Julia 2.0 :)

### CODE REVIEW

**Comment by davidanthoff** [comment]
@chkwon Bump, any chance you could merge this and tag a new version?

Also CC @odow, not sure whether you can merge things here?

File: Project.toml [hunk 1]
```diff
@@ -11,11 +11,11 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
...
 
 [compat]
-JuMP = "0.21"
-PATHSolver = "1.1"
+JuMP = "0.22"
+PATHSolver = "1.2"
 NLsolve = "4"
-MathOptInterface = "0.9"
-julia = "1.3, 2"
+MathOptInterface = "0.10"
```
Comment (odow) [inline Project.toml:17]:
```suggestion
MathOptInterface = "0.10, 1"
```

File: Project.toml [hunk 2]
```diff
@@ -11,11 +11,11 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 
 [compat]
-JuMP = "0.21"
-PATHSolver = "1.1"
+JuMP = "0.22"
```
Comment (odow) [inline Project.toml:14]:
```suggestion
JuMP = "0.22, 0.23"
```

**Review by odow — COMMENTED** [review]
I don't think I have permission. But can you try this?

**Comment by chkwon** [comment]
Thanks @davidanthoff. A PR is submitted.

https://github.com/JuliaRegistries/General/pull/56386

**Comment by davidanthoff** [comment]
Thanks much appreciated!

**Pull request merged**